### PR TITLE
CI: Run clippy on stable and use msrv setting in `clippy.toml`

### PR DESCRIPTION
**Pull request by niklasmohrin**
See #834

### CODE REVIEW

File: .github/workflows/CICD.yml [hunk 1]
```diff
@@ -33,11 +32,26 @@ jobs:
       with:
         command: fmt
         args: -- --check
-    - name: Run clippy (on minimum supported rust version to prevent warnings we can't fix)
+    - name: Ensure MSRV is set in `clippy.toml`
+      run: grep "^msrv = \"${{ env.MIN_SUPPORTED_RUST_VERSION }}\"\$" clippy.toml
+    - name: Run clippy
```
Comment (sharkdp) [inline .github/workflows/CICD.yml:37]: Is there a way to pass `msrv` via the command line to clippy? This way, we could get rid of the chek and we wouldn't need to clutter our root directory.

Comment (niklasmohrin) [reply .github/workflows/CICD.yml:37]: I think we can have it as

```rust
#![clippy::msrv = "1.30.0"]
```

**Comment by niklasmohrin** [comment]
It's been a couple of days since I have been active here :sweat_smile: Please see my edited review comment response (don't know if GH pings everyone on comment edit, thus this message :) )

File: .github/workflows/CICD.yml (hunk 1 above)
Comment (sharkdp) [reply .github/workflows/CICD.yml:37]: Good point. Thank you.

**Review by sharkdp — APPROVED** [review]

**Pull request merged**